p8e-util/src/main/kotlin/io/p8e/crypto/Pen.kt

```diff
@@ -64,19 +64,61 @@ class Pen(
             .orThrow { IllegalStateException("can't verify signature - public cert may not match private key.") }
     }
 
-    override fun sign(): ByteArray = signature.sign()
-
-    override fun update(data: ByteArray) { signature.update(data) }
+    override fun sign(): ByteArray {
+        signature.update(aggregatedData)
```

Comment:
Same as the verify, if the signature input stream continues to push data into the update method of the signature instance, the signing/verification cannot properly verify the objects in the below implementation. Keeping in mind that we had to extract the below implementation out of the Signature library so that SmartKey is able to sign the data accordingly under its own Sign api.